File: app/models/link.rb
# frozen_string_literal: true

# == Schema Information
#
# Table name: links
#
#  id               :bigint           not null, primary key
#  name             :string
#  slug             :string
#  website          :string
#  description      :text
#  link_category_id :bigint           not null
#  created_at       :datetime         not null
#  updated_at       :datetime         not null
#
class Link < ApplicationRecord
  belongs_to :link_category
  slug :name
end

File: src/VegDex.Core/Entities/Link.cs
namespace VegDex.Core.Entities;

public class Link : Entity
{
    public LinkCategory Category { get; set; }
    public string? Description { get; set; }
    public int LinkCategoryId { get; set; }
    public string Name { get; set; }
    public string Slug { get; set; }
    public string Website { get; set; }
    public static Link Create(
        int linkId,
        string name,
        string website,
        LinkCategory linkCategory,
        string? description = null)
    {
        var link = new Link
        {
            Id = linkId,
            Category = linkCategory,
            Name = name,
            Website = website,
            Description = description
        };
        return link;
    }
}
